expression/builtin_compare.go

```diff
@@ -382,38 +383,94 @@ func temporalWithDateAsNumEvalType(argTp *types.FieldType) (argEvalType types.Ev
 	return
 }
 
-// GetCmpTp4MinMax gets compare type for GREATEST and LEAST and BETWEEN
-func GetCmpTp4MinMax(args []Expression) (argTp types.EvalType) {
-	datetimeFound, isAllStr := false, true
-	cmpEvalType, isStr, isTemporalWithDate := temporalWithDateAsNumEvalType(args[0].GetType())
-	if !isStr {
-		isAllStr = false
+func aggregateType(args []Expression) *types.FieldType {
```

Comment:
Is this dup with https://github.com/pingcap/tidb/blob/master/types/field_type.go#L67